lib/ansible/modules/cloud/google/gcp_compute_disk.py

```diff
@@ -449,7 +451,33 @@ def create(module, link, kind):
 
 
 def update(module, link, kind):
-    module.fail_json(msg="Disk cannot be edited")
+    auth = GcpSession(module, 'compute')
+    return wait_for_operation(module, auth.put(link, resource_to_request(module)))
+
+
+def update_fields(module, request, response):
+    difference = GcpRequest(request).difference(GcpRequest(response))
+    auth = GcpSession(module, 'compute')
+    if difference.get('labels'):
+        auth.post(
+            ''.join([
+                "https://www.googleapis.com/compute/v1/",
+                "projects/{project}/zones/{zone}/disks/{name}/setLabels"
+            ]).format(**module.params),
+            {
+                u'labels': module.params.get('labels')
```

Comment:
You'll also need to send labelFingerprint, here and elsewhere.

Reply:
Do I need to? I currently do not support beta APIs on Ansible, but we list all of the labelFingerprints as beta only.

Reply:
Huh!  Well, you do need to, but that probably represents an error in api.yaml.

Reply:
Disk labels are GA!

Reply:
This doesn't seem to have been fixed.

Reply:
Do we want to fix that in this PR? I didn't know if that would clutter it up too much (we'd have changes in P/C/A/T from doing that)

Reply:
Probably best to do it - otherwise you're submitting code that doesn't work.  :(